test/fixtures/cookbooks/test/recipes/default.rb

```diff
@@ -1,3 +1,6 @@
+apt_update 'update' do
+end.run_action(:update) if platform_family?('debian')
```

Comment:
Yep. Because of the way the recipe is written, if it's not done during compile time (and before the include_recipe) the recipe resources happen first. I could modify how the recipe runs but there is probably reasons why they are done the way they are.